distributed/client.py

```diff
@@ -1512,6 +1512,7 @@ def submit(
         pure : bool (defaults to True)
             Whether or not the function is pure.  Set ``pure=False`` for
             impure functions like ``np.random.random``.
+            See "Pure Functions by Default" in :doc:`client` for more details.
```

Comment:
What do you think about linking directly to this section?

Reply:
I think it's preferable to link directly to the section, but I actually don't know how to do it with `:doc` directives, and I thought it would be a deviation from this repo's style to do something like this:

```text
`Pure Functions by Default <./client.html#pure-functions-by-default>`_
```

If you know how to link to an anchor with a `:doc:` directive, could you show me how? The word "doc" is especially hard to search for in Sphinx-related documentation :stuck_out_tongue:

Reply:
I'm not sure if there's a way to do this with the `:doc:` directive. However something: 

```diff
diff --git a/distributed/client.py b/distributed/client.py
index cd60ace2..9d0f2ec1 100644
--- a/distributed/client.py
+++ b/distributed/client.py
@@ -1512,7 +1512,7 @@ class Client:
         pure : bool (defaults to True)
             Whether or not the function is pure.  Set ``pure=False`` for
             impure functions like ``np.random.random``.
-            See "Pure Functions by Default" in :doc:`client` for more details.
+            See :ref:`pure-functions` for more details.
         workers : string or iterable of strings
             A set of worker addresses or hostnames on which computations may be
             performed. Leave empty to default to all workers (common case)
@@ -1632,7 +1632,7 @@ class Client:
         pure : bool (defaults to True)
             Whether or not the function is pure.  Set ``pure=False`` for
             impure functions like ``np.random.random``.
-            See "Pure Functions by Default" in :doc:`client` for more details.
+            See :ref:`pure-functions` for more details.
         workers : string or iterable of strings
             A set of worker hostnames on which computations may be performed.
             Leave empty to default to all workers (common case)
diff --git a/docs/source/client.rst b/docs/source/client.rst
index 508d8349..2679164c 100644
--- a/docs/source/client.rst
+++ b/docs/source/client.rst
@@ -108,6 +108,7 @@ which are used for larger and smaller result sets respectively.

 For more information see the page on :doc:`Managing Computation <manage-computation>`.

+.. _pure-functions:

 Pure Functions by Default
 -------------------------
```

should work (we use this lots of other places throughout the docs)

Reply:
oooo nice, that works!

![image](https://user-images.githubusercontent.com/7608904/112868303-83b83300-9081-11eb-89ac-c5a72ab8b55c.png)

changed in https://github.com/dask/distributed/pull/4644/commits/7a207260b9f31efd0c0a5d7d36ccbf836a519135